stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/StreamFileUtil.kt

```diff
@@ -54,4 +59,127 @@ public object StreamFileUtil {
             null
         }
     }
+
+    /**
+     * Creates a Stream cache directory if one doesn't exist already.
+     *
+     * @param context The [Context] necessary to perform
+     * file operations.
+     * @param onError Called when an error occurs.
+     *
+     * @return The Stream cache directory if it exists or its creation
+     * was successful, false otherwise.
+     */
+    @Suppress("TooGenericExceptionCaught")
+    private fun getOrCreateStreamCacheDir(
+        context: Context,
+        onError: (exception: Exception) -> Unit = {},
+    ): File? {
+        return try {
+            val streamCacheDir = File(context.cacheDir, streamCacheDirName)
+            streamCacheDir.mkdir()
+
+            streamCacheDir
+        } catch (e: Exception) {
+            onError(e)
+            null
+        }
+    }
+
+    /**
+     * Deletes all the content contained within the
+     * Stream cache directory.
+     *
+     *  @param context The [Context] necessary to perform
+     *  file operations.
+     * @param onError Called when an error occurs.
+     *
+     * @return True if the cache was cleared, false otherwise.
+     */
+    @Suppress("TooGenericExceptionCaught")
+    public fun clearStreamCache(
+        context: Context,
+        onError: (exception: Exception) -> Unit = {},
+    ): Boolean {
+        return try {
+            val directory = File(context.cacheDir, streamCacheDirName)
+
+            directory.listFiles()?.forEach {
+                it.delete()
+            }
+
+            true
+        } catch (e: Exception) {
+            onError(e)
+            false
+        }
+    }
```

Comment:
public fun clearStreamCache(context: Context): Boolean = File(context.cacheDir, streamCacheDirName).deleteRecursively()

Reply:
I don't necessarily want to delete the folder as well.
It's a folder that might end up being used relatively often and I'd prefer it stay here even when it's empty.

If you see any downside with that approach, please let me know.

Reply:
If for some reason a folder is created in this folder and it is not empty, this clear method will fails and it won¡t be able to clean the cache at all.

Reply:
Reviewing our code, I can see we are silencing the exception.
On this method, you are catching whatever Exception (Which exception could happen here? Why don't we have a more clear one?) but, because the `onError` function has a default empty implementation we are not handling it.
Whenever we have an error on this method, we are not going to be notified and think all works fine.

Reply:
Adopted `deleteRecursively()` and a switch to using `Result<T>` in order to make exception handling clearer, as discussed in the meeting.